src/components/Filter/Filter.jsx

```diff
@@ -0,0 +1,23 @@
+export default function Filter({
+  pokedexTypes,
+  selectedType,
+  handleFilterChange,
+}) {
+  return (
+    <div>
+      <select
+        value={selectedType}
+        onChange={(event) => handleFilterChange(event.target.value)}
+      >
+        <option key="all" value="all">
+          All
+        </option>
+        {pokedexTypes.map((type) => (
```

Comment:
The error popping up for this map might be related to how you're initializing the pokedexTypes state in Compendium.jsx. It might be having a problem with using an empty object there since you can't map through an object. I'd start by console logging what comes out of that fetch call you use to set that state